scripts/update-ssh.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -euo pipefail
+
+user=$1
+
+# generate clean authorized_keys file from current sources list. This
+# will drop keys that have been removed from this repo or from github
+tmp=$(mktemp)
+
+# start with explicit keys from this repo
+if test -f "keys/$user"; then
```

Comment:
Yes, this is designed to be run from repo/, via ./scripts/upsate-ssh.sh, so CWD is still repo. I have tested it manually that it works. I'm gonna add a key for me on the TPP server and add it the repo in keys/bloodearnest though, so I can check.

Reply:
We _could_ make these scripts agnostic as to CWD using script-relative paths. Might not be worth it though.

For reference the line I've been cargo-culting about the place is:
```sh
script_dir="$( unset CDPATH && cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Reply:
I'd be tempted to just error if they are not invoked from repo/. e.g.

    test -d $PWD/.git || { echo "This script must be run from the repo's root directory"; exit 1 }

Then we can safely assume things.

Reply:
Yeah, throwing an error early sounds like a better plan. Could combine this with `BASH_SOURCE` for a slightly more robust check than looking for `.git` but almost certainly not worth it.

Reply:
great - so, this PR works as advertised, happy to put any enhancements in a follow-up PR.
FWIW I'm also in favour of throwing an error rather than trying to become agnostic, let's discuss it in the follow-up.